mono/metadata/custom-attrs.c

```diff
@@ -1710,3 +1711,89 @@ mono_reflection_get_custom_attrs_data_checked (MonoObject *obj, MonoError *error
 
 	return result;
 }
+
+/**
+ * mono_assembly_metadata_foreach_custom_attr:
+ * @assembly: the assembly to iterate over
+ * @func: the function to call for each custom attribute
+ * @user_data: passed to @func
+ *
+ * Calls @func for each custom attribute type on the given assembly until @func returns FALSE.
+ * Everything is done using low-level metadata APIs, so it is safe to use during assembly loading.
```

Comment:
The usual convention is to continue until the find function returns TRUE, not FALSE.